### ISSUE TITLE
Encoder/decoder for percent-encoding (URL/URI encoding)

### ISSUE THREAD

**Issue by hhromic** [mikefarah/yq#1529]
**Please describe your feature request.**
(( First of all, thank you very much for developing `yq`. This is really a great swiss army knife for YAML! 💯 ))

It would be quite handy for `yq` to support [percent-encoding](https://en.wikipedia.org/wiki/URL_encoding) (aka URL/URI encoding) as an encoder/decoder, similar to the existing `@base64` encoder/decoder pair. For instance, `jq` supports this as a [string formatter](https://stedolan.github.io/jq/manual/#Builtinoperatorsandfunctions):
```
$ jq -n '{key:@uri "uri=\("some/value")"}'
```
```json
{
  "key": "uri=some%2Fvalue"
}
```

This is specially handy when constructing URLs from dynamic values. A hypothetical example:
```
$ DB_PASSWORD='super/secret' yq -n '.dbconfig.dsn="mysql://user:" + (env(DB_PASSWORD) | @uri) + "@host:port/dbname"'
```
```yaml
dbconfig:
  dsn: mysql://user:super%2Fsecret@host:port/dbname
```

**Describe alternatives you've considered**
Within `yq`, it doesn't seem to be possible to perform percent-encoding. Using other tools, `jq` supports encoding like shown above, therefore the current workaround is to use `jq` to encode the value and then pass it to `yq` to insert into a YAML document.

**Comment by hhromic** [comment]
Looks like Go has built-in batteries to encode/decode percent-encoding in `net/url` :)
* https://pkg.go.dev/net/url#QueryEscape
* https://pkg.go.dev/net/url#QueryUnescape

**Comment by mikefarah** [comment]
Nice find - should be pretty easy to add that in 👍🏼

**Comment by hhromic** [comment]
Wow, nice you already implemented this! 💯 
Thanks a lot! I will give it a try as soon as this addition is released!

**Comment by mikefarah** [comment]
Released in v4.31.1 :)

**Comment by hhromic** [comment]
Just tested the released binary:
```
$ DB_PASSWORD='super/secret' yq -n '.dbconfig.dsn="mysql://user:" + (env(DB_PASSWORD) | @uri) + "@host:port/dbname"'
dbconfig:
  dsn: mysql://user:super%2Fsecret@host:port/dbname
```
Works exactly like proposed! :)
Thanks!
